drivers/gpio/gpio_silabs_siwx917_uulp.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_siwx917_gpio_uulp
+
+#include "sl_si91x_driver_gpio.h"
+#include "sl_status.h"
+
+#include <errno.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/irq.h>
+
+/* Zephyr GPIO header must be included after driver, due to symbol conflicts
+ * for GPIO_INPUT and GPIO_OUTPUT between preprocessor macros in the Zephyr
+ * API and struct member register definitions for the SiWx917 device.
+ */
```

Comment:
It's kind of risky, and it would be better if it wasn't a problem. But as long as the inclusion order stays as documented in the comment, it's safe. The macro shadows the register name, and we always want to use the macro, not the register.